### ISSUE TITLE
[Story] Update Dev Container to Ubuntu 22.04

### ISSUE THREAD

**Issue by tschaffter** [Sage-Bionetworks/sage-monorepo#1842]
### What projects is this story for?

OpenChallenges

### As a user, I want

NA

### Description

This is a second attempt to update the dev container image from Debian to Ubuntu. The first attempt was documented in #1806.

What has changed?

I'm work on enabling Sage Monorepo to support R in #1683. I noticed that installing packages from source was slow and could be speed up by installing the binaries from another repo, e.g. a POSIT repo. When using Debian and a POSIT repo for Ubuntu Focal (no repo available for Debian), some packages fail to install and so they must be installed from source, which add about 2-3 minutes to the setup time for this R project.

Adopting Ubuntu 22.04 would increase the DX for R developers.

### Have you linked this story to a GitHub Project?

- [X] I have linked this story to a GitHub Project and set its metadata.

**Comment by tschaffter** [comment]
Today I can successfully build the image without the `curl` issue reported in #1806. 🙃

**Comment by tschaffter** [comment]
## Python 3.10 is the default version shipped with Ubuntu 22.04

I bumped into this issue the first time I evaluated Ubuntu 22.04 and then created #1809 to address the issue. A quicker solution for now would be to install Python 3.9.2 with Pyenv in the Dockerfile of the devcontainer.

```console
>  NX   Successfully ran target prepare-java for 17 projects (20s)
 
   View logs and investigate cache misses at https://cloud.nx.app/runs/SH9eTsqfQH



    ✖  nx run openchallenges-notebook:prepare-python
       The currently activated Python version 3.10.6 is not supported by the project (3.9.2).
       Trying to find and use a compatible version. 
       
       Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.
       Warning: run-commands command "poetry install" exited with non-zero status code

    ✖  nx run schematic-api-client-python:prepare-python
       The currently activated Python version 3.10.6 is not supported by the project (3.9.2).
       Trying to find and use a compatible version. 
       
       Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.
       Warning: run-commands command "poetry install" exited with non-zero status code

    ✖  nx run schematic-api:prepare-python
       The specified Python version (3.10.6) is not supported by the project (3.10.0).
       Please choose a compatible version or loosen the python constraint specified in the pyproject.toml file.
       
        >  NX   ERROR: Something went wrong in run-commands - Command failed: poetry env use 3.10.0
       
          Pass --verbose to see the stacktrace.
       

    ✖  nx run schematic-notebook:prepare-python
       The currently activated Python version 3.10.6 is not supported by the project (3.9.2).
       Trying to find and use a compatible version. 
       
       Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.
       Warning: run-commands command "poetry install" exited with non-zero status code

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target prepare-python for 4 projects (2s)
 
    ✔    0/4 succeeded [0 read from cache]
 
    ✖    4/4 targets failed, including the following:
         - nx run openchallenges-notebook:prepare-python
         - nx run schematic-api-client-python:prepare-python
         - nx run schematic-api:prepare-python
         - nx run schematic-notebook:prepare-python
```

**Comment by tschaffter** [comment]
The issue is that running `python --version` in a Python project folder and in an Nx task gives a different output.

When running the cmd in the project folder:

```console
vscode@376fb986fa12:/workspaces/sage-monorepo/apps/schematic/api$ python --version
Python 3.10.0
```

When running the cmd as part of an Nx task:

```json
    "prepare-python": {
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          "python --version"
        ],
        "cwd": "apps/schematic/api",
        "parallel": false
      }
```

```console
vscode@376fb986fa12:/workspaces/sage-monorepo/apps/schematic/api$ nx prepare-python

> nx run schematic-api:prepare-python

Python 3.10.6
```
